controllers/operatorscaler_controller.go

```diff
@@ -0,0 +1,364 @@
+	crdNameNoobaa             = "noobaas.noobaa.io"
+	crdNamePrometheus         = "prometheuses.monitoring.coreos.com"
+	crdNameAlertmanager       = "alertmanagers.monitoring.coreos.com"
+	crdNameFlashSystemCluster = "flashsystemclusters.odf.ibm.com"
+
+	// OLM pkg Names
+	pkgNameCephcsiOperator       = "cephcsi-operator"
+	pkgNameMcgOperator           = "mcg-operator"
+	pkgNameOcsClientOperator     = "ocs-client-operator" //nolint:unused
+	pkgNameOcsOperator           = "ocs-operator"
+	pkgNameOdfCsiAddonsOperator  = "odf-csi-addons-operator"
+	pkgNameOdfDependencies       = "odf-dependencies"        //nolint:unused
+	pkgNameOdfOperator           = "odf-operator"            //nolint:unused
+	pkgNameOdfPrometheusOperator = "odf-prometheus-operator" //nolint:unused
+	pkgNameRecipe                = "recipe"                  //nolint:unused
+	pkgNameRookCephOperator      = "rook-ceph-operator"
+
+	pkgNameIbmStorageOdfOperator = "ibm-storage-odf-operator"
+)
+
+var (
+	kindStorageCluster = &metav1.PartialObjectMetadata{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "ocs.openshift.io/v1",
+			Kind:       "StorageCluster",
+		},
+	}
+	kindCephCluster = &metav1.PartialObjectMetadata{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "ceph.rook.io/v1",
+			Kind:       "CephCluster",
+		},
+	}
+	kindNoobaa = &metav1.PartialObjectMetadata{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "noobaa.io/v1alpha1",
+			Kind:       "NooBaa",
+		},
+	}
+	kindPrometheus = &metav1.PartialObjectMetadata{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "monitoring.coreos.com/v1",
+			Kind:       "Prometheus",
+		},
+	}
+	kindAlertmanager = &metav1.PartialObjectMetadata{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "monitoring.coreos.com/v1",
+			Kind:       "Alertmanager",
+		},
+	}
+	kindFlashSystemCluster = &metav1.PartialObjectMetadata{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "odf.ibm.com/v1alpha1",
+			Kind:       "FlashSystemCluster",
+		},
+	}
+
+	crdNameList = []string{
+		crdNameStorageCluster,
+		crdNameCephCluster,
+		crdNameNoobaa,
+		crdNamePrometheus,
+		crdNameAlertmanager,
+		crdNameFlashSystemCluster,
+	}
+
+	crdNameToKindMapping = map[string]*metav1.PartialObjectMetadata{
+		crdNameStorageCluster:     kindStorageCluster,
+		crdNameCephCluster:        kindCephCluster,
+		crdNameNoobaa:             kindNoobaa,
+		crdNamePrometheus:         kindPrometheus,
+		crdNameAlertmanager:       kindAlertmanager,
+		crdNameFlashSystemCluster: kindFlashSystemCluster,
+	}
+
+	kindToPkgNamesMapping = map[*metav1.PartialObjectMetadata][]string{
+		kindStorageCluster:     {pkgNameOcsOperator},
+		kindCephCluster:        {pkgNameRookCephOperator, pkgNameCephcsiOperator, pkgNameOdfCsiAddonsOperator},
+		kindNoobaa:             {pkgNameMcgOperator},
+		kindPrometheus:         {pkgNameOdfPrometheusOperator},
+		kindAlertmanager:       {pkgNameOdfPrometheusOperator},
+		kindFlashSystemCluster: {pkgNameIbmStorageOdfOperator},
```

Comment:
This does not make sense as a map. The only place you are using it is as part of a for loop. 
An list will be better, or even preferably collapse the info into the previous map values then run on `maps.Values` in the loop

Reply:
Why doesn’t it make sense as a map? We need this mapping because we have to scale up these CSVs based on the kind.

Reply:
Changed it to slice as discussed